Data/Primitive/PrimRef.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE RoleAnnotations #-}
+{-# LANGUAGE UnboxedTuples #-}
+{-# LANGUAGE Unsafe #-}
+
+-- | Note: Edward Kmett wrote everything in this module. 
+module Data.Primitive.PrimVar
+  ( 
+  -- * Primitive References
+    PrimVar(..)
+  , newPrimVar
+  , newPinnedPrimVar
+  , newAlignedPinnedPrimVar
+  , readPrimVar
+  , writePrimVar
+  , primVarContents
+  , primVarToMutablePrimArray
+  , casInt
+  , fetchAddInt
+  , fetchSubInt
+  , fetchAndInt
+  , fetchNandInt
+  , fetchOrInt
+  , fetchXorInt
+  , atomicReadInt
+  , atomicWriteInt
+  ) where
+
+import Control.Monad.Primitive
+import Data.Primitive
+import GHC.Exts
+import GHC.Ptr (castPtr)
+
+--------------------------------------------------------------------------------
+-- * Primitive References
+--------------------------------------------------------------------------------
+
+newtype PrimVar s a = PrimVar (MutablePrimArray s a)
+
+type role PrimVar nominal nominal
+
+-- | Create a primitive reference.
```

Comment:
```suggestion
-- | Create a 'PrimVar'.
```
In general, I'd replace `primitive reference` by `'PrimVar'`.

Reply:
This is not really consistent with the rest of the library.

Reply:
It's consistent with the documentation of `Data.Primitive.MutVar`. At least it should talk about primitive *references*, since we decided to call the type `PrimVar`. Besides, I don't think what the rest of the library does is really important for this kind of things...